src/test/java/io/confluent/maven/resolver/KafkaFetchTest.java

```diff
@@ -0,0 +1,32 @@
+package io.confluent.maven.resolver;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResult;
+import org.eclipse.aether.version.Version;
+import org.eclipse.aether.util.version.GenericVersionScheme;
+
+public class KafkaFetchTest {
+    private static final String CCS = "ccs-kafka";
+    private static final String constrainText = "[6.0.1-1, 6.0.2-1)";
+    private GenericVersionScheme genericVersionScheme = new GenericVersionScheme();
+    private ResolveKafkaVersionRangeMojo resolve = new ResolveKafkaVersionRangeMojo();
+    
+    private static final String ONECCS = "6.0.1-1-ccs";
+    private static final String TWELVECE = "6.0.1-12-ce";
+
+    private final VersionRangeRequest request = new VersionRangeRequest();
+
+    @Test
+    public void testCECCSkafka() throws Exception {
```

Comment:
I think this is the critical one where we want the non-lexicographic behavior, but we should get better coverage by validating at least a CE case as well.

Reply:
Also, I don't think you are really testing to make sure the sort is non-lexicographical. can you make a test with version 6.0.0-100-ce and 6.0.0-2-ce to make sure you get build 100 as the highest. This test is essentially only making sure that when you resolve ccs you don't include ce versions, but it doesn't ensure that you got the correct version/highest version of ccs.

Reply:
Actually I did make some other testcases, but then I find the `getHighestVersion` method is just return whatever the top entry in the Version list, which means the actual sorting happen at when you make a VersionRangeRequest. The request seems bit outscope this plugin. I did not find a good way to mock it, maybe I can try some simple IT tests by make some very simple artifacts.